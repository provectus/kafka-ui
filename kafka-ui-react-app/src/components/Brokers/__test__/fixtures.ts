import { BrokersLogdirs } from 'generated-sources';

export const brokersPayload = [
  { id: 1, host: 'b-1.test.kafka.amazonaws.com', port: 9092 },
  { id: 2, host: 'b-2.test.kafka.amazonaws.com', port: 9092 },
];

export const clusterStatsPayload = {
  brokerCount: 2,
  activeControllers: 1,
  onlinePartitionCount: 138,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 239,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [
    { brokerId: 0, segmentSize: 334567, segmentCount: 245 },
    { brokerId: 1, segmentSize: 12345678, segmentCount: 121 },
  ],
  version: '2.2.1',
};

const partition = {
  broker: 2,
  offsetLag: 0,
  partition: 2,
  size: 0,
};
const topics = {
  name: '_confluent-ksql-devquery_CTAS_NUMBER_OF_TESTS_59-Aggregate-Aggregate-Materialize-changelog',
  partitions: [partition],
};

export const brokerLogDirsPayload: BrokersLogdirs[] = [
  {
    error: 'NONE',
    name: '/opt/kafka/data-0/logs',
    topics: [
      {
        ...topics,
        partitions: [partition, partition, partition],
      },
      topics,
      {
        ...topics,
        partitions: [],
      },
    ],
  },
];
