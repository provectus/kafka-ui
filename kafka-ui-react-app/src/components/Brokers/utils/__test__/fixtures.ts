import { BrokerMetrics } from 'generated-sources';

export const brokerMetricsPayload: BrokerMetrics = {
  segmentSize: 23,
  segmentCount: 23,
  metrics: [
    {
      name: 'TotalFetchRequestsPerSec',
      labels: {
        canonicalName:
          'kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics',
      },
      value: 10,
    },
    {
      name: 'ZooKeeperRequestLatencyMs',
      value: 11,
    },
    {
      name: 'RequestHandlerAvgIdlePercent',
    },
  ],
};
export const transformedBrokerMetricsPayload =
  '{"segmentSize":23,"segmentCount":23,"metrics":[{"name":"TotalFetchRequestsPerSec","labels":{"canonicalName":"kafka.server:name=TotalFetchRequestsPerSec,topic=_connect_status,type=BrokerTopicMetrics"},"value":10},{"name":"ZooKeeperRequestLatencyMs","value":11},{"name":"RequestHandlerAvgIdlePercent"}]}';
