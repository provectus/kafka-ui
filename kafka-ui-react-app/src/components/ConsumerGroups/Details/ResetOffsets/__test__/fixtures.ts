export const expectedOutputs = {
  EARLIEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'EARLIEST',
    topic: '__amazon_msk_canary',
  },
  LATEST: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: undefined,
        partition: 0,
      },
    ],
    resetType: 'LATEST',
    topic: '__amazon_msk_canary',
  },
  OFFSET: {
    partitions: [0],
    partitionsOffsets: [
      {
        offset: '10',
        partition: 0,
      },
    ],
    resetType: 'OFFSET',
    topic: '__amazon_msk_canary',
  },
};
