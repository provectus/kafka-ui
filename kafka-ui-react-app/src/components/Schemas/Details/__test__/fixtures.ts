import { SchemaSubject, SchemaType } from 'generated-sources';
import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

export const versionPayload = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];
export const versionEmptyPayload = [];

export const jsonSchema: SchemaSubject = {
  subject: 'test',
  version: '15',
  id: 1,
  schema:
    '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.JSON,
};

export const protoSchema: SchemaSubject = {
  subject: 'test_proto',
  version: '1',
  id: 2,
  schema:
    'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
  compatibilityLevel: 'BACKWARD',
  schemaType: SchemaType.PROTOBUF,
};
