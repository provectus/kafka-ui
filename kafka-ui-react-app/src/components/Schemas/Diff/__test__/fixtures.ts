import { SchemaSubject, SchemaType } from 'generated-sources';

export const versions: SchemaSubject[] = [
  {
    subject: 'test',
    version: '3',
    id: 3,
    schema:
      'syntax = "proto3";\npackage com.indeed;\n\nmessage MyRecord {\n  int32 id = 1;\n  string name = 2;\n}\n',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.PROTOBUF,
  },
  {
    subject: 'test',
    version: '2',
    id: 2,
    schema:
      '{"type":"record","name":"MyRecord2","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
  {
    subject: 'test',
    version: '1',
    id: 1,
    schema:
      '{"type":"record","name":"MyRecord1","namespace":"com.mycompany","fields":[{"name":"id","type":"long"}]}',
    compatibilityLevel: 'BACKWARD',
    schemaType: SchemaType.JSON,
  },
];
