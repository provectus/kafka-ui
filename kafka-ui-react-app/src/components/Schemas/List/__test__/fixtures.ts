import {
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
} from 'redux/reducers/schemas/__test__/fixtures';

const schemas = [
  schemaVersion1,
  schemaVersion2,
  schemaVersionWithNonAsciiChars,
];

export const schemasPayload = {
  pageCount: 1,
  schemas,
};

export const schemasEmptyPayload = {
  pageCount: 1,
  schemas: [],
};
