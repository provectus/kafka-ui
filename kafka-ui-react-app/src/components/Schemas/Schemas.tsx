import React from 'react';
import { Route, Routes } from 'react-router-dom';
import {
  clusterSchemaEditRelativePath,
  clusterSchemaNewRelativePath,
  clusterSchemaSchemaDiffRelativePath,
  RouteParams,
} from 'lib/paths';
import List from 'components/Schemas/List/List';
import Details from 'components/Schemas/Details/Details';
import New from 'components/Schemas/New/New';
import Edit from 'components/Schemas/Edit/Edit';
import DiffContainer from 'components/Schemas/Diff/DiffContainer';

const Schemas: React.FC = () => {
  return (
    <Routes>
      <Route index element={<List />} />
      <Route path={clusterSchemaNewRelativePath} element={<New />} />
      <Route path={RouteParams.subject} element={<Details />} />
      <Route path={clusterSchemaEditRelativePath} element={<Edit />} />
      <Route
        path={clusterSchemaSchemaDiffRelativePath}
        element={<DiffContainer />}
      />
    </Routes>
  );
};

export default Schemas;
