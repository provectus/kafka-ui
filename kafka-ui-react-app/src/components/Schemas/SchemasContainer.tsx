import { connect } from 'react-redux';
import { RootState } from 'redux/interfaces';
import { fetchSchemasByClusterName } from 'redux/actions';
import { getIsSchemaListFetched } from 'redux/reducers/schemas/selectors';
import Schemas from './Schemas';

const mapStateToProps = (state: RootState) => ({
  isFetched: getIsSchemaListFetched(state),
});

const mapDispatchToProps = {
  fetchSchemasByClusterName,
};

export default connect(mapStateToProps, mapDispatchToProps)(Schemas);
