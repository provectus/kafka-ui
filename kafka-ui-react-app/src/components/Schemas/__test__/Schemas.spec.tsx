import React from 'react';
import Schemas from 'components/Schemas/Schemas';
import { render, WithRoute } from 'lib/testHelpers';
import {
  clusterSchemaEditPath,
  clusterSchemaNewPath,
  clusterSchemaPath,
  clusterSchemasPath,
  getNonExactPath,
} from 'lib/paths';
import { screen, waitFor } from '@testing-library/dom';
import fetchMock from 'fetch-mock';
import { schemaVersion } from 'redux/reducers/schemas/__test__/fixtures';

const renderComponent = (pathname: string) =>
  render(
    <WithRoute path={getNonExactPath(clusterSchemasPath())}>
      <Schemas />
    </WithRoute>,
    { initialEntries: [pathname] }
  );

const clusterName = 'secondLocal';

const SchemaCompText = {
  List: 'List',
  Details: 'Details',
  New: 'New',
  Edit: 'Edit',
};

jest.mock('components/Schemas/List/List', () => () => (
  <div>{SchemaCompText.List}</div>
));
jest.mock('components/Schemas/Details/Details', () => () => (
  <div>{SchemaCompText.Details}</div>
));
jest.mock('components/Schemas/New/New', () => () => (
  <div>{SchemaCompText.New}</div>
));
jest.mock('components/Schemas/Edit/Edit', () => () => (
  <div>{SchemaCompText.Edit}</div>
));

describe('Schemas', () => {
  beforeEach(() => {
    fetchMock.getOnce(`/api/clusters/${clusterName}/schemas`, [schemaVersion]);
  });
  afterEach(() => fetchMock.restore());
  it('renders List', async () => {
    renderComponent(clusterSchemasPath(clusterName));
    await waitFor(() =>
      expect(screen.queryByText(SchemaCompText.List)).toBeInTheDocument()
    );
  });
  it('renders New', async () => {
    renderComponent(clusterSchemaNewPath(clusterName));
    await waitFor(() =>
      expect(screen.queryByText(SchemaCompText.New)).toBeInTheDocument()
    );
  });
  it('renders Details', async () => {
    renderComponent(clusterSchemaPath(clusterName, schemaVersion.subject));
    await waitFor(() =>
      expect(screen.queryByText(SchemaCompText.Details)).toBeInTheDocument()
    );
  });
  it('renders Edit', async () => {
    renderComponent(clusterSchemaEditPath(clusterName, schemaVersion.subject));
    await waitFor(() =>
      expect(screen.queryByText(SchemaCompText.Edit)).toBeInTheDocument()
    );
  });
});
