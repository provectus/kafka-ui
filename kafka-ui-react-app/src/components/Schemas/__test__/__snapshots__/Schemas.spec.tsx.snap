// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schemas Container View Initial state matches snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/ui/clusters/:clusterName/schemas"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/ui/clusters/:clusterName/schemas/:subject/latest"
  />
</Switch>
`;

exports[`Schemas Container View when page is loading matches snapshot 1`] = `<PageLoader />`;
